Automattic/woocommerce-payments
# Refresh account balance amounts on the Payments Overview screen upon successful instant deposit

Fixes #7829
Fixes #7841

#### Changes proposed in this Pull Request

This PR fixes stale account balance data upon a successful instant deposit.

It does this by invalidating the response from the `deposits/overview-all` response upon a successful instant deposit request.

This will:

1. trigger the Payments → Overview Account Balances card to fetch the balances 
2. show updated balance amounts, reflecting the instant deposit balance transaction
3. hide the **Deposit funds** button if the server API response deems the store no longer eligible (fixes #7841)

See screen recording of solution:

https://github.com/Automattic/woocommerce-payments/assets/3147296/5306b45a-af3d-4bef-92ed-e6527433911b


#### Testing instructions





1. Ensure your store is eligible for instant deposits and has an instant balance available to deposit
2. Navigate to Payments → Overview. Pay attention to the available and pending account balances.
3. Click **Deposit funds**
4. On success, ensure that the available and pending balances have been updated to reflect the instant deposit transaction.

An e2e test would be suitable for this, but we currently have no e2e tests for instant deposits.

-------------------


**Post merge**
